### PR TITLE
Update API docs

### PR DESCRIPTION
**Pull request by pinheadmz**
This is a pretty massive update to the api-docs section of the website.

The first few commits are ports from `bcoin-org.github.io` for style and UI fixes and improvements. There's a few typos that were cleared up and missing API endpoints that pre-date the hsd fork form bcoin. The existing wallet auction RPC page had a copypasted tpyo throughout (`NodeClient` instead of `WalletClient`)

~The big changes are:~

~Adding new node http commands (https://github.com/handshake-org/hsd/pull/189) and adding wallet auction http commands (https://github.com/handshake-org/hsd/pull/163)~

### CODE REVIEW

File: src/api-docs/source/index.html.md [hunk 1]
```diff
@@ -90,8 +92,20 @@ const client = new NodeClient(clientOptions);
...
 
+
 <aside class="notice">
-You must replace <code>api-key</code> with your personal API key.
+You must replace <code>api-key</code> with your own, strong API key.<br>
+<br>
+A good way to generate a strong key is to use the <code>bcrypto</code> module that is installed as a 
+dependency for <code>hsd</code>. From your hsd directory (or anywhere, if <code>bcrypto</code> is installed globally), run:<br>
+<code>node -e "bcrypto=require('bcrypto'); console.log(bcrypto.random.randomBytes(32).toString('hex'))"</code><br>
+Which will generate and output a securley random, 32-byte hex string.<br>
```
Comment (boymanjor) [inline src/api-docs/source/index.html.md]: s/securley/securely/

File: src/api-docs/source/includes/_node.md [hunk 2]
```diff
@@ -324,4 +400,179 @@ Broadcast a transaction by adding it to the node's mempool. If mempool verificat
...
+
+> The above command returns JSON structured like this:
+
+```json
+{
+  "rate": 13795
+}
+```
+
+Estimate the fee required (in Dollarydoos per kB) for a transaction to be confirmed by the network within a targeted number of blocks (default 1).
```
Comment (boymanjor) [inline src/api-docs/source/includes/_node.md]: nit: I'm not sure I've seen dollarydoos capitalized.

File: src/api-docs/source/index.html.md (hunk 1 above)
Comment (boymanjor) [reply src/api-docs/source/index.html.md]: unrelated nit: there is a typo in the commit message.

**Comment by pinheadmz** [comment]
Broke off the two bigger chunks into separate PRs #42 and #43

**Pull request merged**